tapgarden/planter.go

```diff
@@ -1698,6 +1774,38 @@ func (c *ChainPlanter) fundBatch(ctx context.Context, params FundParams,
 	return nil
 }
 
+// matchPsbtToGroupReq attempts to match a signed group virtual PSBT to a
+// corresponding group key request. This is done by comparing the tapscript root
+// in the PSBT to the tapscript root in the group key request.
+func matchPsbtToGroupReq(psbt psbt.Packet,
+	groupReqs []asset.GroupKeyRequest) fn.Option[asset.GroupKeyRequest] {
+
+	// If there is only one group request, we can assume that the PSBT
+	// matches it.
+	//
+	// TODO(ffranr): This is a workaround because the PSBT does not contain
```

Comment:
I think we should add some minimal information to the group key minting PSBT. This can be used by UIs to display additional information w.r.t what is being requested for mint.

Reply:
We should be able to match on the `input[0].PrevOutpoint`.
We can generate the genesis TX from the `GroupKeyRequest` using this:
```
tx, err := req.BuildGroupVirtualTx(&tapscript.GroupTxBuilder{})
```